receiver/hostmetricsreceiver/internal/scraper/processscraper/process_scraper.go

```diff
@@ -0,0 +1,352 @@
+
+	"go.opentelemetry.io/collector/component/componenterror"
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/internal/processor/filterset"
+)
+
+// scraper for Process Metrics
+type scraper struct {
+	config    *Config
+	startTime pdata.TimestampUnixNano
+	includeFS filterset.FilterSet
+	excludeFS filterset.FilterSet
+
+	getProcessHandles func() (processHandles, error)
+}
+
+type processMetadata struct {
+	pid      int32
+	name     string
+	handle   processHandle
+	username string
+	cmdline  string
+}
+
+type processTimes struct {
+	*processMetadata
+	times *cpu.TimesStat
+}
+
+type processMemoryInfo struct {
+	*processMetadata
+	memoryInfo *process.MemoryInfoStat
+}
+
+type processIoCounters struct {
+	*processMetadata
+	ioCounters *process.IOCountersStat
+}
+
+// newProcessScraper creates a Process Scraper
+func newProcessScraper(cfg *Config) (*scraper, error) {
+	scraper := &scraper{config: cfg, getProcessHandles: getProcessHandlesInternal}
+
+	var err error
+
+	if len(cfg.Include.Names) > 0 {
+		scraper.includeFS, err = filterset.CreateFilterSet(cfg.Include.Names, &cfg.Include.Config)
+		if err != nil {
+			return nil, fmt.Errorf("error creating process include filters: %v", err)
+		}
+	}
+
+	if len(cfg.Exclude.Names) > 0 {
+		scraper.excludeFS, err = filterset.CreateFilterSet(cfg.Exclude.Names, &cfg.Exclude.Config)
+		if err != nil {
+			return nil, fmt.Errorf("error creating process exclude filters: %v", err)
+		}
+	}
+
+	return scraper, nil
+}
+
+// Initialize
+func (s *scraper) Initialize(_ context.Context) error {
+	bootTime, err := host.BootTime()
+	if err != nil {
+		return err
+	}
+
+	s.startTime = pdata.TimestampUnixNano(bootTime)
+	return nil
+}
+
+// Close
+func (s *scraper) Close(_ context.Context) error {
+	return nil
+}
+
+// ScrapeMetrics
+func (s *scraper) ScrapeMetrics(ctx context.Context) (pdata.MetricSlice, error) {
+	_, span := trace.StartSpan(ctx, "processscraper.ScrapeMetrics")
+	defer span.End()
+
+	var errors []error
+	metrics := pdata.NewMetricSlice()
+
+	processes, err := s.getProcesses()
+	if err != nil {
+		errors = append(errors, err)
+	}
+
+	if err = scrapeAndAppendCPUUsageMetric(metrics, s.startTime, processes); err != nil {
+		errors = append(errors, err)
+	}
+
+	if err = scrapeAndAppendMemoryUsageMetric(metrics, s.startTime, processes); err != nil {
+		errors = append(errors, err)
+	}
+
+	if err = scrapeAndAppendDiskBytesMetric(metrics, s.startTime, processes); err != nil {
+		errors = append(errors, err)
+	}
+
+	if len(errors) > 0 {
+		return metrics, componenterror.CombineErrors(errors)
+	}
+
+	return metrics, nil
+}
+
+// getProcesses returns a slice of processMetadata, including handles,
+// for all currently running processes. If errors occur obtaining
+// information for some processes, an error will be returned, but any
+// processes that were successfully obtained will still be returned.
+func (s *scraper) getProcesses() ([]*processMetadata, error) {
```

Comment:
Would try making doing processMetadata by value and seeing if it improves benchmark.